File: pruebasProyecto/gate.cpp
#include "gate.h"

using namespace std;


// Constructor
    Gate::Gate(int _id, const string& _location,
         bool _availability, const string& _boardingHour)
        : id(_id), location(_location),
          availability(_availability), boardingHour(_boardingHour) {
    }
    Gate::Gate(){}

    // Métodos get
    Flight Gate::getFlightAssigned()  { return *flightAssigned; }
    int Gate::getId()  { return id; }
    string Gate::getLocation()  { return location; }
    bool Gate::isAvailable()  { return availability; }
    string Gate::getBoardingHour()  { return boardingHour; }
    vector<Flight*> Gate::getFlightsRecord()  { return flightsRecord; }

    // Métodos set
    void Gate::setFlightAssigned( Flight *_flightAssigned) {flightAssigned = _flightAssigned; }
    void Gate::setId(int _id) { id = _id; }
    void Gate::setLocation(const string& _location) { location = _location; }
    void Gate::setAvailability(bool _availability) { availability = _availability; }
    void Gate::setBoardingHour(const string& _boardingHour) { boardingHour = _boardingHour; }
    void Gate::addFlightsRecord(Flight *_flight) {flightsRecord.push_back(_flight);}

    void Gate::print() {
    cout << "Gate Information:" << endl;
    cout << "ID: " << id << endl;
    cout << "Location: " << location << endl;
    cout << "Availability: " << (availability ? "Available" : "Not Available") << endl;
    cout << "Boarding Hour: " << boardingHour << endl;

    // Imprime la información del vuelo asignado
    if (flightAssigned != nullptr) {
        cout << "Flight Assigned Information:" << endl;
        flightAssigned->print();
    }

    // Imprime la información de los vuelos registrados
    cout << "Flights Record Information:" << endl;
    for (int i = 0; i < flightsRecord.size(); ++i) {
        cout << "Flight " << i + 1 << ":" << endl;
        flightsRecord[i]->print();
    }
}

File: proyectoPython/gate.py
class Gate:
    def __init__(self, ident = None, location = None, availability = None, boarding_hour = None):
        self.ident = ident
        self.location = location
        self.availability = availability
        self.boarding_hour = boarding_hour
        self.flight_assigned = None
        self.flights_record = []

    def set_flight_assigned(self, flight_assigned):
        self.flight_assigned = flight_assigned

    def set_ident(self, ident):
        self.ident = ident

    def set_location(self, location):
        self.location = location

    def set_availability(self, availability):
        self.availability = availability

    def set_boarding_hour(self, boarding_hour):
        self.boarding_hour = boarding_hour

    def get_flight_assigned(self):
        return self.flight_assigned

    def get_ident(self):
        return self.ident

    def get_location(self):
        return self.location

    def is_available(self):
        return self.availability

    def get_boarding_hour(self):
        return self.boarding_hour

    def get_flights_record(self):
        return self.flights_record

    def add_flights_record(self, flight):
        self.flights_record.append(flight)